data/src/main/java/com/linkedin/data/schema/DataSchemaParser.java

```diff
@@ -0,0 +1,114 @@
+package com.linkedin.data.schema;
+
+import java.io.InputStream;
+import java.io.Reader;
+import java.util.List;
+
+/**
+ * A schema parser is used to parse data schemas for a particular source representation.
+ */
+public interface DataSchemaParser {
```

Comment:
Okay, cool. I believe `SchemaParser` is also taken.  How about `PegasusSchemaParser` ?